features/promotion/applying_catalog_promotions/reapplying_catalog_promotions_after_editing_its_channels.feature

```diff
@@ -0,0 +1,39 @@
+@applying_catalog_promotions
+Feature: Reapplying catalog promotions after editing its channels
+    In order to be attracted to products by its current prices
+    As a Visitor
+    I want to see current discounted prices of products in proper channel
+
+    Background:
+        Given the store operates on a channel named "Web-US" with hostname "web-us"
+        And the store operates on another channel named "Web-GB" with hostname "web-gb"
+        And the store has a "T-Shirt" configurable product
+        And this product has "PHP T-Shirt" variant priced at "$20.00" in "Web-US" channel
+        And "PHP T-Shirt" variant priced at "$30.00" in "Web-GB" channel
+        And this product is available in "Web-US" channel and "Web-GB" channel
+        And there is a catalog promotion "Winter sale" available in "Web-US" channel that reduces price by "30%" and applies on "PHP T-shirt" variant
+        And I am currently in the "Web-US" channel
+
+    @todo
+    Scenario: Removing applied catalog promotion after removing its channel
+        Given this catalog promotion is no longer available in "Web-US" channel
```

Comment:
Won't the behaviour be better covered if this step looks like:
```suggestion
        Given the administrator removed "Web-US" channel from the "Winter sale" catalog promotion
```
?
And then it would be performed in the API/UI instead of removing the channel from entity in setup context. Wdyt?

Reply:
for me it is ok, but IMHO this new step should be `when` because, channel removing is an action that causes changes in the system, so `when admin remove a channel, then promotions are reapplied

Reply:
From the technical PoV, it makes sense to conduct an actual **action** rather than programmatically removing the channel from the entity.

*But* on the other hand, these scenarios are written from the **Visitor** perspective. So the visitor does not care *how* did it happen that the promotion is no longer available (either the administrator edits it, there is a problem in the system or whatever else). They just see the result of the action. additionally, it would be tricky (but, of course, possible) to switch between Admin (to conduct an action) and Visitor (to see the result) context, in both UI and API implementation.

So, summing this up: I'm sure Visitor should have no knowledge about the administrator's actions, so I would leave it as it is right know 🖖

Reply:
After small discussion, I decided to change the actor of these scenarios